### PR TITLE
Add public examples to std.file + enable DScanner check

### PR DESCRIPTION
**Pull request by wilzbach**

### CODE REVIEW

File: std/file.d [hunk 1]
```diff
@@ -1213,14 +1284,14 @@ if (isConvertibleToString!R)
     auto currTime = Clock.currTime();
 
     write(deleteme, "a");
-    scope(exit) { assert(exists(deleteme)); remove(deleteme); }
+    scope(exit) deleteme.exists && deleteme.remove;
```
Comment (JackStouffer) [inline std/file.d]: why remove the assert?

File: std/file.d [hunk 2]
```diff
@@ -1585,6 +1675,31 @@ if (isConvertibleToString!R)
...
+    SysTime modificationTime = SysTime(DateTime(2011, 10, 4, 0, 0, 30), 10.hnsecs);
+    setTimes(deleteme, accessTime, modificationTime);
+
+    SysTime accessTimeResolved, modificationTimeResolved;
+    getTimes(deleteme, accessTimeResolved, modificationTimeResolved);
+
+    assert(accessTime == accessTimeResolved);
+    assert(modificationTime == modificationTimeResolved);
+
+    import std.datetime : Clock, seconds;
```
Comment (JackStouffer) [inline std/file.d]: Everything before this line isn't related to `timeLastModified` and should be removed IMO

File: std/file.d (hunk 1 above)
Comment (wilzbach) [reply std/file.d]: Removes throws and exception if the file doesn't exist

File: std/file.d [hunk 3]
```diff
@@ -1492,6 +1563,28 @@ if (isInputRange!R && !isInfinite!R && isSomeChar!(ElementEncodingType!R) &&
...
+    setTimes(deleteme, accessTime, modificationTime);
+
+    SysTime accessTimeResolved, modificationTimeResolved;
+    getTimes(deleteme, accessTimeResolved, modificationTimeResolved);
+
+    assert(accessTime == accessTimeResolved);
+    import std.stdio;
+    modificationTime.writeln;
+    modificationTimeResolved.writeln;
+    assert(modificationTime == modificationTimeResolved);
```
Comment (wilzbach) [inline std/file.d]: This is failing on OSX. Maybe OSX doesn't have so fine-grained timestamps?

Comment (wilzbach) [reply std/file.d]: Looks like it:

```
> modificationTime.writeln;
> modificationTimeResolved.writeln;
2018-Oct-04 00:00:30.000001
2018-Oct-04 00:00:30
```

**Pull request merged**